homeassistant/components/tado/climate.py

```diff
@@ -500,26 +514,26 @@ def _async_update_zone_data(self):
         self._current_tado_swing_mode = self._tado_zone_data.current_swing_mode
 
     @callback
-    def _async_update_zone_callback(self):
+    def _async_update_zone_callback(self) -> None:
         """Load tado data and update state."""
         self._async_update_zone_data()
         self.async_write_ha_state()
 
     @callback
-    def _async_update_home_data(self):
+    def _async_update_home_data(self) -> None:
         """Load tado geofencing data into zone."""
         self._tado_geofence_data = self._tado.data["geofence"]
 
     @callback
-    def _async_update_home_callback(self):
+    def _async_update_home_callback(self) -> None:
         """Load tado data and update state."""
         self._async_update_home_data()
         self.async_write_ha_state()
 
-    def _normalize_target_temp_for_hvac_mode(self):
+    def _normalize_target_temp_for_hvac_mode(self) -> None:
         # Set a target temperature if we don't have any
         # This can happen when we switch from Off to On
-        if self._target_temp is None:
+        if self._target_temp is None or isinstance(self._target_temp, float):
```

Comment:
that doesn't look right to me. You are declaring `self._target_temp` to be a `float` but override it now, if it is. Are you missing a `not`?

Reply:
It was indeed missing. Made some more optimization, since it's chunky and MyPy was complain about the the operator making a > comparison with None. The last patch should mitigate this and make it less redundant.